### PR TITLE
feat: handle assistant sync task

### PR DESCRIPTION
**Pull request by luka-nextcloud**
### 📝 Summary

* Resolves: #4686

- update assistant name space
- add insert button for assistant sync task result
- use notifications for triggering fetch instead of polling

#### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
B | A

### 🚧 TODO

- [ ] ...

### 🏁 Checklist

- [ ] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [ ] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [ ] Documentation ([README](https://github.com/nextcloud/text/blob/main/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -41,7 +41,7 @@
 use OCA\Text\Middleware\SessionMiddleware;
 use OCA\Text\Notification\Notifier;
 use OCA\Text\Service\ConfigService;
-use OCA\TPAssistant\Event\BeforeAssistantNotificationEvent;
+use OCA\TpAssistant\Event\BeforeAssistantNotificationEvent;
```
Comment (juliushaertl) [inline lib/AppInfo/Application.php:44]: Can you add the class to the psalm stub file to fix the failing psalm action?

File: src/components/Assistant.vue [hunk 2]
```diff
@@ -213,7 +214,7 @@ export default {
 	},
 	computed: {
 		showAssistant() {
-			return !this.$isRichWorkspace && !this.$isPublic && window?.OCA?.TPAssistant?.openAssistantForm
+			return !this.$isRichWorkspace && !this.$isPublic && window?.OCA?.TpAssistant?.openAssistantForm
```
Comment (juliushaertl) [inline src/components/Assistant.vue:217]: @julien-nc Was there a casing change in the API here? Is it bound to a Nextcloud release?

**Review by juliushaertl — COMMENTED** [review]
Small comments, otherwise this looks good and works (with https://github.com/nextcloud/server/pull/43005 for testing)

**Comment by juliushaertl** [comment]
@luka-nextcloud Could you also take the time to add cypress coverage for this part? We can enable the testing app for this in the github action.

**Comment by juliushaertl** [comment]
Tested and works function wise 👍 

@julien-nc I noticed some odd behaviour with https://github.com/nextcloud/server/pull/43005 where having multiple text processing providers enabled leads to the async one alwas being used in the assistant modal. I haven't touched any configuration, though the AI settings show the synchronous one as a default:

Screenshot 2024-01-23 at 19 48 22

**Comment by luka-nextcloud** [comment]
> @luka-nextcloud Could you also take the time to add cypress coverage for this part? We can enable the testing app for this in the github action.

I've added it, please check.

File: .github/workflows/cypress.yml [hunk 3]
```diff
@@ -41,6 +41,13 @@ jobs:
           ref: ${{ matrix.server-versions }}
           path: apps/viewer
 
+      - name: Checkout assistant
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          repository: nextcloud/assistant
+          ref: 'main'
+          path: apps/assistant
```
Comment (juliushaertl) [inline .github/workflows/cypress.yml:49]: Need to ensure that both assistant and testing app are also enabled in line 151 when setting up the Nextcloud instance to be used in the cypress tests

File: cypress/entrypoint.sh [hunk 4]
```diff
@@ -19,8 +19,10 @@ if [ $? -eq 0 ]; then
         php /var/www/html/occ user:add --password-from-env user2
         php /var/www/html/occ app:enable viewer
         php /var/www/html/occ app:enable text
+        php /var/www/html/occ app:enable assistant
```
Comment (juliushaertl) [inline cypress/entrypoint.sh:22]: I think this was for some way of running isolated cypress tests locally, but never used that script myself. @max-nextcloud Do you happen to know if we still make use of that?

**Comment by juliushaertl** [comment]
@luka-nextcloud Could you check why the cypress tests fail there?

File: src/components/Assistant.vue (hunk 2 above)
Comment (julien-nc) [reply src/components/Assistant.vue:217]: Sorry for the late answer. This was and will stay OCA.TPAssistant.

**Comment by julien-nc** [comment]
> I noticed some odd behaviour with https://github.com/nextcloud/server/pull/43005 where having multiple text processing providers enabled leads to the async one alwas being used in the assistant modal. I haven't touched any configuration, though the AI settings show the synchronous one as a default:

@juliushaertl Thanks, I'll check that. Maybe there's a mismatch between the default value displayed in the settings frontend and the actual default selected provider when creating a task.

**Review by juliushaertl — APPROVED** [review]

File: cypress/entrypoint.sh (hunk 4 above)
Comment (max-nextcloud) [reply cypress/entrypoint.sh:22]: I don't remember it. Looking at the log of that file indicates that it was introduced by @susnux in f62c29be.

**Comment by juliushaertl** [comment]
Pushed a small adjustment for the github action that should make cypress pass.

File: cypress/entrypoint.sh (hunk 4 above)
Comment (susnux) [reply cypress/entrypoint.sh:22]: Was just for local tests, but nowadays we could use `@nextcloud/cypress/docker` which also works locally :)

**Pull request merged**